let CustomErrorHandler,JwtService;_575‍.x([["default",()=>_575‍.o]]);_575‍.w("../services/CustomErrorHandler",[["default",["CustomErrorHandler"],function(v){CustomErrorHandler=v}]]);_575‍.w("../services/JwtSertice",[["default",["JwtService"],function(v){JwtService=v}]]);


const auth = async (req, res, next) => {
  let authHeader = req.headers.authorization;

  if (!authHeader) {
    return next(CustomErrorHandler.unAuthorized());
  }
  const token = authHeader.split(" ")[1];
  _575‍.g.console.log(token);
  
  try {
    const {_id, role} = await JwtService.verify(token);
    const user = {_id, role};
    req.user = user;
    next();
  } catch (error) {
    return next(CustomErrorHandler.unAuthorized());
  }
};

_575‍.d(auth);
